src/content/ccip/tutorials/ccipreceive-gaslimit.mdx

```diff
@@ -0,0 +1,495 @@
+
+1. Send three CCIP messages with different number of iterations:
+
+   ```bash
+   npx hardhat run scripts/testing/sendCCIPMessages.ts --network avalancheFuji
+   ```
+
+   Example output:
+
+   ```text
+   $ npx hardhat run scripts/testing/sendCCIPMessages.ts --network avalancheFuji
+   Approving 0x0b9d5D9136855f6FEc3c0993feE6E9CE8a297846 for 0x32A24e40851E19d1eD2a7E697d1a38228e9388a3. Allowance is 115792089237316195423570985008687907853269984665640564039457584007913129639935. Signer 0x9d087fC03ae39b088326b67fA3C788236645b717...
+   115792089237316195423570985008687907853269984665640564039457584007913129639935n
+
+   Number of iterations 0 - Gas limit: 5685 - Message Id: 0xf23b17366d69159ea7d502835c4178a1c1d1d6325edf3d91dca08f2c7a2900f7
+   Number of iterations 50 - Gas limit: 16190 - Message Id: 0x4b3a97f6ac959f67d769492ab3e0414e87fdd9c143228f9c538b22bb695ca728
+   Number of iterations 99 - Gas limit: 26485 - Message Id: 0x37d1867518c0f8c54ceb0c5507b46b8d44c6c53864218f448cba0234f8de867a
+   ```
+
+1. Open the [CCIP explorer](https://ccip.chain.link), search each message by its ID, and wait for each message to be successfully transmitted (`Status` in the explorer: `Success`).
+
+For the example above, here are the destination transaction hashes:
+
+    | Message id                                                         | Ethereum Sepolia transaction hash                                  |
+    | ------------------------------------------------------------------ | ------------------------------------------------------------------ |
+    | 0xf23b17366d69159ea7d502835c4178a1c1d1d6325edf3d91dca08f2c7a2900f7 | 0xf004eb6dab30b3cfb9d1d631c3f9832410b8d4b3179e65b85730563b67b1e689 |
+    | 0x4b3a97f6ac959f67d769492ab3e0414e87fdd9c143228f9c538b22bb695ca728 | 0xf004eb6dab30b3cfb9d1d631c3f9832410b8d4b3179e65b85730563b67b1e689 |
+    | 0x37d1867518c0f8c54ceb0c5507b46b8d44c6c53864218f448cba0234f8de867a | 0xf004eb6dab30b3cfb9d1d631c3f9832410b8d4b3179e65b85730563b67b1e689 |
+
+**Note** that the Ethereum Sepolia transaction hash is the same for all the messages. This is because CCIP batched the messages.
+
+### Check the actual gas usage
+
+1. Open [Tenderly](https://dashboard.tenderly.co) and Search for the [destination transaction hash](https://dashboard.tenderly.co/tx/sepolia/0xf004eb6dab30b3cfb9d1d631c3f9832410b8d4b3179e65b85730563b67b1e689).
+
+1. Search for `_callWithExactGasSafeReturnData` with a payload containing your messageId (without `0x`). [Example](https://dashboard.tenderly.co/tx/sepolia/0xf004eb6dab30b3cfb9d1d631c3f9832410b8d4b3179e65b85730563b67b1e689?trace=0.8.4.2) for `0xf23b17366d69159ea7d502835c4178a1c1d1d6325edf3d91dca08f2c7a2900f7`.
+
+1. Below it, you will find the call trace from the Router to your Receiver contract.[Call trace example](https://dashboard.tenderly.co/tx/sepolia/0xf004eb6dab30b3cfb9d1d631c3f9832410b8d4b3179e65b85730563b67b1e689?trace=0.8.4.2.0).
+
+1. Click on the debugger and you'll get the gas details:
```

Comment:
Assuming you mean this Debugger tab?

<img width="227" alt="Screenshot 2024-03-18 at 1 20 14 AM" src="https://github.com/smartcontractkit/documentation/assets/5498502/758f43ff-e149-4ab3-b382-cab2d3f83bb7">

Not suggesting we add screenshots of Tenderly UI here, but we should refer to UI elements very precisely for clarity and accessibility.